### PR TITLE
[IMP] mrp: adapt propagation mechanism

### PR DESCRIPTION
**Pull request by ajf-odoo**
No longer consider 'produce_delay' when searching for a manufacturing order to merge within.

task: 3430776

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mrp/models/stock_rule.py [hunk 1]
```diff
@@ -45,38 +45,49 @@ def _should_auto_confirm_procurement_mo(self, p):
...
-
             mo = self.env['mrp.production']
-            mto_route = self.env['stock.warehouse']._find_global_route('stock.route_warehouse0_mto', _('Replenish on Order (MTO)'))
-            if rule.route_id != mto_route and procurement.origin != 'MPS':
-                gpo = rule.group_propagation_option
-                group = (gpo == 'fixed' and rule.group_id) or \
-                        (gpo == 'propagate' and 'group_id' in procurement.values and procurement.values['group_id']) or False
+            orderpoint = procurement.values.get('orderpoint_id')
+            gpo = rule.group_propagation_option
+            if (procurement.origin != 'MPS') and (orderpoint or gpo != 'propagate'):
```
Comment (amoyaux) [inline addons/mrp/models/stock_rule.py:52]: I would add a function that define if merge is allow or not. Could be useful for hooks

Comment (ajf-odoo) [reply addons/mrp/models/stock_rule.py]: done

File: addons/mrp/models/stock_move.py [hunk 2]
```diff
@@ -487,6 +487,12 @@ def _key_assign_picking(self):
         keys = super(StockMove, self)._key_assign_picking()
         return keys + (self.created_production_id,)
 
+    def _search_picking_for_assignation_domain(self):
+        res = super()._search_picking_for_assignation_domain()
+        if self.created_production_id:
+            res = expression.AND([res[1:], ['|', res[0], ('move_ids.created_production_id', '=', self.created_production_id.id)]])
```
Comment (svs-odoo) [inline addons/mrp/models/stock_move.py:493]: This method is overrided in `stock_picking_batch`, adding a bunch of `'&'` in the domain and making this code very prompt to traceback

Comment (svs-odoo) [reply addons/mrp/models/stock_move.py:493]: After some tests, it seems it happens only when `mrp` is installed after `stock_picking_batch` and the server isn't rebooted meanwhile, so the call order is: `mrp` -> call super in `stock_picking_batch` -> call super in `stock` and then, when we are at this line, `res[0]` is `'&'`, so the second parameter for `expression.AND` is `['|', '&', ('move_ids.created_production_id', '=', self.created_production_id.id)]` which causes the traceback

**Pull request closed without merging**